simtools/configuration/commandline_parser.py

```diff
@@ -231,26 +236,218 @@ def initialize_simulation_model_arguments(self, model_options):
         ----------
         model_options: list
-            Options to be set: "telescope", "site"
+            Options to be set: "telescope", "site", "layout", "layout_file"
         """
-        if model_options is not None:
-            _job_group = self.add_argument_group("simulation model")
-            if "site" in model_options or "telescope" in model_options:
-                _job_group.add_argument(
-                    "--site", help="site (e.g., North, South)", type=self.site, required=False
-                )
-            if "telescope" in model_options:
-                _job_group.add_argument(
-                    "--telescope",
-                    help="telescope model name (e.g., LSTN-01, SSTS-design, ...)",
-                    type=self.telescope,
-                )
+        if model_options is None:
+            return
+
+        _job_group = self.add_argument_group("simulation model")
+        _job_group.add_argument(
+            "--model_version",
+            help="model version",
+            type=str,
+            default="Released",
+        )
+        if any(
+            option in model_options for option in ["site", "telescope", "layout", "layout_file"]
+        ):
+            self._add_model_option_site(_job_group)
+
+        if "telescope" in model_options:
             _job_group.add_argument(
-                "--model_version",
-                help="model version",
+                "--telescope",
+                help="telescope model name (e.g., LSTN-01, SSTS-design, ...)",
+                type=self.telescope,
+            )
+
+        if "layout" in model_options or "layout_file" in model_options:
+            _job_group = self._add_model_option_layout(_job_group, "layout_file" in model_options)
+
+    def initialize_simulation_configuration_arguments(self, simulation_configuration):
+        """
+        Initialize default arguments for simulation configuration and simulation software.
+
+        Parameters
+        ----------
+        simulation_configuration: list
+            List of simulation software configuration parameters.
+        """
+        if simulation_configuration is None:
+            return
+
+        if "software" in simulation_configuration:
+            self._initialize_simulation_software()
+        if "corsika_configuration" in simulation_configuration:
+            self._initialize_simulation_configuration()
+            self._initialize_shower_configuration()
+
+    def _initialize_simulation_software(self):
+        """Initialize simulation software arguments."""
+        _software_group = self.add_argument_group("simulation software")
+        _software_group.add_argument(
+            "--simulation_software",
+            help="Simulation software steps.",
+            type=str,
+            choices=["corsika", "simtel", "corsika_simtel"],
+            required=True,
+            default="corsika_simtel",
+        )
+
+    def _initialize_simulation_configuration(self):
```

Comment:
These arguments are all very specific to the simulation applications. By putting them here we do not have an argument anymore in the future against putting all the other parameters such as the magnetic field strength, particle interaction model,... Maybe this is ok for now, but I think we should be more flexible in the sense that we should not need to list all possible parameters manually in the code.

Reply:
Perhaps a better solution would be to use the DB and check whether a passed parameter exists and use it in case positive?

Reply:
We separate those parameters which might 

1. change per production (e.g., the energy range and the number of showers)
2.  those which do not change (e.g., the hadronic interaction model or geomagnetic field value)

1. is configured via the command line and 2. is part of the simulation model (to be added; add this point in data data/parameters/corsika_parameters.yml

I do think this makes sense.